pkgs/by-name/fe/fex/package.nix

```diff
@@ -2,31 +2,58 @@
   fetchFromGitHub,
   lib,
   llvmPackages,
+  stdenv,
   cmake,
   ninja,
   pkg-config,
-  gitMinimal,
   qt5,
   python3,
+  xxHash,
+  fmt,
+  nasm,
 }:
 
-llvmPackages.stdenv.mkDerivation (finalAttrs: rec {
+llvmPackages.stdenv.mkDerivation (finalAttrs: {
   pname = "fex";
   version = "2503";
 
   src = fetchFromGitHub {
     owner = "FEX-Emu";
     repo = "FEX";
-    tag = "FEX-${version}";
-    hash = "sha256-NnYod6DeRv3/6h8SGkGYtgC+RRuIafxoQm3j1Sqk0mU=";
-    fetchSubmodules = true;
+    tag = "FEX-${finalAttrs.version}";
+
+    hash = "sha256-yjxwwqnI0jartl97zGfqP9GJSQ6HRwZ6ZJUWPsFsvUw=";
+
+    leaveDotGit = true;
+    postFetch = ''
+      cd $out
+      git reset
+
+      # Only fetch required submodules
+      git submodule update --init --depth 1 \
+        External/Vulkan-Headers \
+        External/drm-headers \
+        External/jemalloc \
+        External/jemalloc_glibc \
+        External/robin-map \
+        External/vixl \
+        Source/Common/cpp-optparse \
+        External/Catch2 \
+        External/xbyak
```

Comment:
Why are we not using our versions of those libraries?

Reply:
That is the long term goal, here is the upstream issue tracking this: https://github.com/FEX-Emu/FEX/issues/2996

This PR is a move into this direction, by eliminating the `External/fmt` and `External/xxhash` submodules, and using our own libraries instead.

I believe `catch2` should also be an easy win, and I do plan on looking into swapping it for our own package in a follow up PR. (But I don't want to scope creep this one further, it's already large enough.)